ft232/mpsse.py

```
@@ -0,0 +1,555 @@
+        Data.append(self.__PinValues[0])
+        Data.append(self.__PinDirections[0])
+        ## Checks if the board has more than 8 Pins
+        if self.__highByteExists():
+            Data.append(self.__pinConfigCommand(MPSSE.CONFIG_HIGH_PINS, MPSSE.SET_PIN_CONFIG))        
+            Data.append(self.__PinValues[1])
+            Data.append(self.__PinDirections[1])
+        self.__Ft232.write(bytes(Data))
+
+    def __getPinConfig(self):
+        """ Gets the actual PinValues from the device as a list.
+        
+        Returns
+        -------
+        list
+            The PinValues as a list of bytes
+        """
+        Command = []
+        Command.append(self.__pinConfigCommand(MPSSE.CONFIG_LOW_PINS, MPSSE.GET_PIN_CONFIG))
+        ## Checks if the board has more than 8 Pins
+        if self.__highByteExists():
+            Command.append(self.__pinConfigCommand(MPSSE.CONFIG_HIGH_PINS, MPSSE.GET_PIN_CONFIG))
+        self.__Ft232.write(bytes(Command))
+        return self.__Ft232.read(len(Command))
+
+    def __setPinDirection(self, Pin, InputOutputPin):
+        """ Make a Pin an input or output. This does not actually set the pin physically. Use _setPinDirection() or _setPinsDirection() instead. Otherwise call _setPinConfig() after using this function to set the direction physically.
+        Parameters
+        ----------
+        Pin : int, 4 - 15
+            The Pin whose direction should be set.
+        
+        InputOutputPin : GPIO.INPUT / GPIO.OUTPUT
+            The direction the Pin is set to.
+        """
+        if (Pin >= self.__Ft232.D3) and self.__isPinLowByte(Pin):
+            self.__PinDirections[0] = (self.__PinDirections[0] & ~(1 << Pin)) | (InputOutputPin << Pin)
+        elif self.__isPinHighByte(Pin):
+            self.__PinDirections[1] = (self.__PinDirections[1] & ~(1 << (Pin - 8))) | (InputOutputPin << (Pin - 8))
+        else:
+            print("Pin is out of range")
```

The library should never print. You should probably raise an exception for an error.

---

Yes, i think i missed a few of those prints which should be exceptions.